photutils/detection/findstars.py

```diff
@@ -7,18 +7,20 @@
 import warnings
 import math
 import numpy as np
+import abc
 from astropy.table import Column, Table
 from astropy.utils.exceptions import AstropyUserWarning
 from astropy.stats import gaussian_fwhm_to_sigma
 from .core import _convolve_data, find_peaks
 
 
-__all__ = ['daofind', 'irafstarfind']
+class StarFinder(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def __call__(self, data):
```

Comment:
Yes. Thanks!
I'm leaning towards your suggestion.

Other than the doc updates, how does this look?